test/unit/carrier_test.rb

```diff
@@ -0,0 +1,100 @@
+require 'test_helper'
+
+class CarrierTest < Minitest::Test
+  class ExampleCarrier < Carrier
+    cattr_reader :name
+    @@name = "Example Carrier"
+  end
+
+  def setup
+    @carrier = ExampleCarrier.new
+  end
+
+  def test_find_rates_not_implemented
+    assert_raises NotImplementedError do
+      @carrier.find_rates(nil, nil, nil)
+    end
+  end
+
+  def test_create_shipment_not_implemented
+    assert_raises NotImplementedError do
+      @carrier.create_shipment(nil, nil, nil)
+    end
+  end
+
+  def test_cancel_shipment_not_implemented
+    assert_raises NotImplementedError do
+      @carrier.cancel_shipment(nil)
+    end
+  end
+
+  def test_find_tracking_info_not_implemented
+    assert_raises NotImplementedError do
+      @carrier.find_tracking_info(nil)
+    end
+  end
+
+  def test_maximum_weight
+    assert_equal Quantified::Mass.new(150, :pounds), @carrier.maximum_weight
+  end
+
+  def test_maximum_address_field_length
+    assert_equal 255, @carrier.maximum_address_field_length
+  end
+
+  def test_requirements_empty_array
+    assert_equal [], @carrier.send(:requirements)
+  end
+
+  def test_timestamp_from_business_day_returns_nil_without_a_day
+    assert_nil @carrier.send(:timestamp_from_business_day, nil)
+  end
+
+  def test_save_request
+    request = Object.new
+    assert_nil @carrier.last_request
+    @carrier.send(:save_request, request)
+    assert_equal request, @carrier.last_request
+  end
+
+  def test_timestamp_from_business_day_returns_two_days_in_the_future
+    current = DateTime.civil(2016, 7, 19) # Tuesday
```

Comment:
TIL of `civil`.
Interestingly, [`civil`](http://ruby-doc.org/stdlib-1.9.3/libdoc/date/rdoc/DateTime.html#method-c-civil) is an alias to [`.new`](http://ruby-doc.org/stdlib-1.9.3/libdoc/date/rdoc/DateTime.html#method-c-new).